### PR TITLE
GCI-2306 Implement item summary service

### PR DESCRIPTION
**Pull request by cgregoryCH**
* Implement item summary service 
* Returns mapped item when HTTP success code
* Returns client error when HTTP 404 returned by api
* Returns server error when any other HTTP error code

[GCI-2306](https://companieshouse.atlassian.net/browse/GCI-2306)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -26,12 +26,12 @@
     "gulp-cli": "^2.3.0",
     "http-errors": "~1.6.3",
     "ioredis": "^5.0.4",
+    "js-yaml": "^3.14.0",
     "jade": "~1.11.0",
```
Comment (cgregoryCH) [inline package.json:30]: try adding @types/yaml

File: src/orderitemsummary/MissingImageDeliveryDetailsComponent.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import {AbstractViewComponent} from "../core/AbstractViewComponent";
+import {MissingImageDeliverySummary} from "./MissingImageDeliverySummary";
+import {ViewModel} from "../core/ViewModel";
+
+export class MissingImageDeliveryDetailsComponent extends AbstractViewComponent {
+
+    constructor(private summary: MissingImageDeliverySummary) {
+        super("orderItemSummary/order-item-summary-mid.njk", []);
```
Comment (cgregoryCH) [inline src/orderitemsummary/MissingImageDeliveryDetailsComponent.ts:8]: change to underscores

File: src/orderitemsummary/OrderItemSummaryService.ts [hunk 3]
```diff
@@ -1,12 +1,47 @@
...
+import {OrderItemErrorResponse} from "@companieshouse/api-sdk-node/dist/services/order/order-item/service";
+import {createLogger} from "@companieshouse/structured-logging-node";
+import {Status} from "../core/Status";
 
+@Service()
 export class OrderItemSummaryService {
-    constructor(private factory: OrderItemSummaryFactory = new OrderItemSummaryFactory()) {
+    private static readonly logger = createLogger("OrderItemSummaryService");
+
+    constructor(@Inject("default.client") public apiClientFactory: ApiClientFactory,
```
Comment (cgregoryCH) [inline src/orderitemsummary/OrderItemSummaryService.ts:20]: add condition for stub api client in inject annotation

File: src/orderitemsummary/OrderItemSummaryService.ts [hunk 4]
```diff
@@ -1,12 +1,47 @@
...
+        const response: Result<Item, OrderItemErrorResponse> = await apiClient.orderItem.getOrderItem(request.orderId, request.itemId);
+
+        if (response.isSuccess()) {
+            const mapper = this.factory.getMapper(new MapperRequest(request.orderId, response.value));
+            mapper.map();
+            return {
+                status: Status.SUCCESS,
+                viewModel: mapper.getMappedOrder()
+            };
+        } else if (response.isFailure() && response.value.httpStatusCode === 404) {
```
Comment (cgregoryCH) [inline src/orderitemsummary/OrderItemSummaryService.ts:35]: add truthy check on response errors field

File: test/__mocks__/order.mocks.ts [hunk 5]
```diff
@@ -42,3 +43,25 @@ export const mockMissingImageDeliveryItem: Item = {
...
+            itemId: "MID-123456-123456",
+            companyName: "The Company",
+            companyNumber: "00000000",
+            date: "26 May 2015",
+            type: "AP01",
+            description: "Appointment of Mr Richard John Harris as a director",
+            fee: "£3",
+            backLinkUrl: "/orders-admin/order-summaries/ORD-123456-123456"
+        },
+        template: "orderItemSummary/order-item-summary-mid.njk"
```
Comment (cgregoryCH) [inline test/__mocks__/order.mocks.ts:61]: change to underscores

File: test/orderitemsummary/OrderItemSummaryService.test.ts [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+            expect(mapper.getMappedOrder).toHaveBeenCalled();
+            expect(factory.getMapper).toHaveBeenCalledWith(new MapperRequest( "ORD-123456-123456", mockMissingImageDeliveryItem));
+        });
+
+        it("Returns client error when api returns 404 not found", async () => {
+            // given
+            const response = new Failure<Item, OrderItemErrorResponse>({
+                httpStatusCode: 404
+            });
+
```
Comment (cgregoryCH) [inline test/orderitemsummary/OrderItemSummaryService.test.ts:68]: add mapper mocks back and verify no interactions (toHaveBeenCalledTimes)

**Review by cgregoryCH — COMMENTED** [review]
address pr comments

File: package.json (hunk 1 above)
Comment (cgregoryCH) [reply package.json:30]: yaml requires newer versions of node and npm so sticking with js-yaml for now

**Review by dlgroves-ch — APPROVED** [review]

**Review by sthompsonCH — APPROVED** [review]

**Pull request merged**